src/transformers/modeling_utils.py

```diff
@@ -942,7 +944,10 @@ def _load_state_dict_into_meta_model(
             if is_fsdp_enabled() or is_deepspeed_zero3_enabled():
                 module, tensor_name = get_module_from_name(model, param_name)
                 value = getattr(module, tensor_name)
-                value = type(value)(value.data.to("cpu"), **value.__dict__)
+                param_to = "cpu"
+                if is_fsdp_enabled() and value.data.device.index != 0:
```

Comment:
I do have a question about the comparison to `0` here and above. Is this still valid if a user does something like `CUDA_VISIBLE_DEVICES=2,3,4,5`? If it's not valid then, is there a better way to compare?

Reply:
The way this should work is that `CUDA_VISIBLE_DEVICES` will select the GPUs that the process will see, but they'll still be mapped with a zero-based index. That is `CUDA_VISIBLE_DEVICES=2,3,4,5` will appear as `cuda:0` through `cuda:3` in PyTorch. We can test to confirm this though!

Reply:
thanks! updating this to use `is_local_dist_rank_0()` to be a bit cleaner.